### PR TITLE
Add message if JavaScript is disabled

### LINKED ISSUE #2719: If no JavaScript is available Gutenberg is returning a blank white page
I am well aware that Gutenberg relies heavily onto JavaScript with React, ESNext and Webpack in particular. But never the less from a Progressive Enhancement standpoint it should never ever be the case with the unavailability of JavaScript that there is no content accessible at all, like seeing a blank page for Gutenberg at the moment (WordPress 4.8.1 and Gutenberg 1.1.0) . :/  There are many scenarios where that might happen, you never know (https://kryogenix.org/code/browser/everyonehasjs.html). I am not saying that the same functionality should be provided like in the presence of JavaScript but the content inside the editor should be visible in any given scenario in my humble opinion.

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Adds a noscript and message if Javascript is disabled.

This also required adding a :not(.no-js) to the fullscreen body class
which hides/shows the rest of the wp-admin chrome.

## How has this been tested?

* Disable Javascript using console settings 
1. Prior to change confirm the page is blank.
2. After change you should see the message

screen shot 2018-11-08 at 11 23 34 am

## Types of changes

See #2719 

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [X] My code follows the accessibility standards. 
- [X] My code has proper inline documentation.

### CODE REVIEW

File: gutenberg.php [hunk 1]
```diff
@@ -29,6 +29,13 @@ function the_gutenberg_project() {
 	global $post_type_object;
 	?>
 	<div class="block-editor gutenberg">
+		<noscript>
+			<div class="error" style='margin-top:32px'><p>
+			<?php
+			echo __( 'The block editor requires Javascript to be enabled.', 'gutenberg' );
```
Comment (ocean90) [inline gutenberg.php]: `echo __()` => `_e()`

File: gutenberg.php [hunk 2]
```diff
@@ -29,6 +29,13 @@ function the_gutenberg_project() {
 	global $post_type_object;
 	?>
 	<div class="block-editor gutenberg">
+		<noscript>
+			<div class="error" style='margin-top:32px'><p>
```
Comment (ocean90) [inline gutenberg.php]: The admin bar has different heights depending on the viewport.

Comment (mkaz) [reply gutenberg.php]: The margin was simply to give some spacing from the bottom of the admin bar, so the error message isn't squished up against the top.

File: gutenberg.php (hunk 1 above)
Comment (mkaz) [reply gutenberg.php]: The `echo __()` function is used in numerous spots in gutenberg.php, I just mirrored it to stay consistent within the same file.

Comment (mkaz) [reply gutenberg.php]: Thumbs down is not really a productive response.
I've switched each of the occurrences to the shorter function.

File: packages/edit-post/src/components/fullscreen-mode/style.scss [hunk 3]
```diff
@@ -1,4 +1,4 @@
-body.is-fullscreen-mode {
```
Comment (swissspidy) [inline packages/edit-post/src/components/fullscreen-mode/style.scss]: See also #11677

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Comment by michelleweber** [comment]
I always like to give a specific next step in an error message when possible. In this case, that would be something like "The Block Editor requires Javascript. Go to your browser's settings and enable Javascript before using it. Learn more." 

Ideally the "learn more" is a link to instructions re: how to enable JS, but if we don't have that, just delete it.

File: gutenberg.php [hunk 4]
```diff
@@ -29,6 +29,13 @@ function the_gutenberg_project() {
...
+			<?php
+			_e( 'The block editor requires Javascript to be enabled.', 'gutenberg' );
```
Comment (chrisvanpatten) [inline gutenberg.php]: JavaScript should be CamelCased!

```suggestion
			_e( 'The block editor requires JavaScript to be enabled.', 'gutenberg' );
```

**Review by chrisvanpatten — CHANGES_REQUESTED** [review]
Small typo!

**Comment by mkaz** [comment]
@michelleweber I've updated with your change, but after revisiting the bug, I think [the suggestion here](https://github.com/WordPress/gutenberg/issues/2719#issuecomment-437435084) by @afercia might be better, adding a to link to the Classic Editor plugin 

The user probably either has JavaScript disabled on purpose, or it not supported by their device, so the plugin instructions might be more helpful.

**Comment by michelleweber** [comment]
Works for me. As long as we give the user some kind of next step, we're good -- I leave it to you to decide what the actual best next step is!

**Comment by swissspidy** [comment]
I know this is now getting a bit more complex, but can we add some more conditions whether the Classic Editor plugin is already installed and whether it's active? Perhaps also whether the user has the capability to install it.

At least a direct link to the classic editor if the plugin is already active would be nice.

File: packages/edit-post/src/components/fullscreen-mode/style.scss [hunk 5]
```diff
@@ -1,4 +1,4 @@
...
+body.is-fullscreen-mode:not(.no-js) {
```
Comment (afercia) [inline packages/edit-post/src/components/fullscreen-mode/style.scss]: rather than `:not(.no-js)` I'd simply use `body.js.is-fullscreen-mode`

**Comment by mkaz** [comment]
@swissspidy I know, so much effort on a rare edge case, but might as well do it properly.

I did consider that about if the plugin is installed, however the user should not get to this page since it is a Gutenberg page and if the plugin is installed they will go straight to the Classic Editor.

I also just had to push up some CSS z-index positioning stuff to make it clickable, no such thing as an quick fix.

**Comment by afercia** [comment]
@mkaz thanks for working on this!

Seems it's not possible to actually click on the link because the `#editor` container overlays the notice 🙂 

@jasmussen is printing out by default `is-fullscreen-mode` via PHP really necessary? I guess there are JS ways to set it before everything is rendered.

File: packages/edit-post/src/components/fullscreen-mode/style.scss (hunk 5 above)
Comment (mkaz) [reply packages/edit-post/src/components/fullscreen-mode/style.scss]: Thank you @afercia !

**Comment by mkaz** [comment]
@afercia can you make sure you are updated to the latest PR, I just pushed up a fix to address the issue of it being clickable. Also, just updated with your CSS suggestion.

**Comment by swissspidy** [comment]
What if the plugin is installed but not active?

**Comment by afercia** [comment]
The link is clickable now, thanks!

File: gutenberg.php [hunk 6]
```diff
@@ -28,6 +28,16 @@
 function the_gutenberg_project() {
 	global $post_type_object;
 	?>
+	<noscript>
+		<div class="error" style='position:absolute;top:32px;z-index:40'><p>
```
Comment (GaryJones) [inline gutenberg.php]: Why use single quotes for the style attribute?

File: gutenberg.php [hunk 7]
```diff
@@ -28,6 +28,16 @@
...
+		<?php
+			printf(
+				__( 'The Block Editor requires JavaScript, please try the <a href="%s">Classic Editor plugin</a>.', 'gutenberg' ),
```
Comment (GaryJones) [inline gutenberg.php]: This will read better as two sentences.
"The Block Editor requires JavaScript. Please try the Classic Editor plugin."

**Comment by gziolo** [comment]
> The Block Editor requires JavaScript. Please try the Classic Editor plugin.

I think that this message is valid in the context of WordPress core. I'm not convinced that it is the way to go when Gutenberg plugin is activated. @mcsf @youknowriad @mtias do you agree?

**Comment by mkaz** [comment]
@mcsf @gziolo - I've updated based on your suggestions.

**Comment by gziolo** [comment]
Thanks @mkaz for updating 👍

File: gutenberg.php [hunk 8]
```diff
@@ -28,6 +28,25 @@
 function the_gutenberg_project() {
 	global $post_type_object;
 	?>
+	<noscript>
+		<div class="error" style="position:absolute;top:32px;z-index:40"><p>
+		<?php
+			// Using Gutenberg as Plugin
+			if ( is_plugin_active( 'gutenberg/gutenberg.php' ) ) {
+				$current_url = esc_url( $_SERVER['REQUEST_URI'] . "&classic-editor" );
```
Comment (mcsf) [inline gutenberg.php]: This (concatenating `"&classic-editor"`) will break if `REQUEST_URI` doesn't carry prior query arguments. Should use `add_query_arg` or equivalent.

Comment (mkaz) [reply gutenberg.php]: I've updated, though I'm not sure if that can happen on the editor page.

Comment (mcsf) [reply gutenberg.php]: > I've updated, though I'm not sure if that can happen on the editor page.

By default, Add New Post leads to `/wp-admin/post-new.php` and Add New Page leads to `/wp-admin/post-new.php?post_type=page`. Both situations are to be expected, so `add_query_arg` is definitely necessary.

**Comment by chrisvanpatten** [comment]
@mkaz Oh sorry, I thought the review would have been auto-dismissed after you incorporated my suggestion. Get that fixed up, GitHub!

Approving in a sec :)

**Review by chrisvanpatten — APPROVED** [review]

**Comment by mkaz** [comment]
Thanks @chrisvanpatten I marked as resolved too, I do think it's a Github bug.

**Pull request merged**